cf/sns.yaml

```diff
@@ -22,24 +22,26 @@ Conditions:
 Resources:
   SQSQueue:
     Type: AWS::SQS::Queue
-    Properties: 
+    Properties:
       ContentBasedDeduplication: true
       DeduplicationScope: "messageGroup"
       FifoQueue: true
       QueueName: !If [isDev, "development-queue.fifo", !Sub "queue-job-${SandboxID}-${Environment}.fifo"]
       FifoThroughputLimit: "perMessageGroupId"
-  SNSTopic: 
+  SNSTopic:
     Type: AWS::SNS::Topic
-    Properties: 
+    Properties:
       KmsMasterKeyId: "alias/aws/sns"
       TopicName: !Sub "work-results-${Environment}-${SandboxID}"
   PipelineResultsSubscription:
     Type: 'AWS::SNS::Subscription'
     Properties:
       TopicArn: !Ref SNSTopic
-      Endpoint: !If [isProd, "https://api.scp.biomage.net/v1/pipelineResults",
-                !If [isDev, "http://host.docker.internal:3000/v1/pipelineResults",
-                !Sub "https://api-${SandboxID}.scp-${Environment}.biomage.net/v1/pipelineResults"]]
+      Endpoint: !Sub [ "${BaseUrl}/v1/pipelineResults", BaseUrl: !If [
+                        isDev, "http://host.docker.internal:3000", !If [
+                        isProd, !Sub ["https://api.${DomainName}", DomainName: !Join ["", [ Fn::ImportValue: !Sub "DomainName-${Environment}" ]]],
+                        !Sub ["https://api-${SandboxID}.${DomainName}", DomainName: !Join ["", [ Fn::ImportValue: !Sub "DomainName-${Environment}" ]]]
+                      ]]]
```

Comment:
Instead of having this ifs every time you want to use `BaseUrl`, you can define a parameter at the top of the file and reuse it in this and the other places.

Reply:
I've tried this - Cloudformation does not allow the use of Fn::ImportValue in parameters or conditions

Reply:
- Can not use Fn::ImportValue in parameter ([ref](https://stackoverflow.com/q/60446398/1940886))
- Can not use Fn::ImportValue in conditions ([ref](https://stackoverflow.com/a/62076636/1940886))